src/lib/io/sol-iio.c

```diff
@@ -1107,10 +1112,12 @@ channel_get_offset(struct sol_iio_channel *channel)
 
 #define GET_OFFSET(_name) \
     do { \
-        r = craft_filename_path(path, sizeof(path), CHANNEL_OFFSET_PATH, \
-            device->device_id, _name); \
-        if (r && sol_util_read_file(path, "%d", &channel->offset) > 0) { \
-            goto end; \
+        if (_name) { \
+            r = craft_filename_path(path, sizeof(path), CHANNEL_OFFSET_PATH, \
+                device->device_id, _name); \
+            if (r && sol_util_read_file(path, "%d", &channel->offset) > 0) { \
+                goto end; \
+            } \
```

Comment:
my point is, if `_name` is NULL this macro will do nothing, then the next instruction (line 1126) will be `channel_get_pure_name()` with the parameter `channel` (please remember that `channel->name` is NULL previous check). Now look the function `channel_get_pure_name()`:
 ```
static char *                                                                                                  
channel_get_pure_name(struct sol_iio_channel *channel)                                                         
{                                                                                                              
    char channel_name[NAME_MAX];                                                                               
    size_t channel_name_len;                                                                                   
    char *channel_pure_name;                                                                                   
    bool modified = false;                                                                                     
                                                                                                               
    strncpy(channel_name, channel->name, sizeof(channel_name) - 1);
    ....
```
As you can see, it will crash on `strncpy()`

So, it seems that you can not deal with `channel->name == NULL` in any case. Probably the check should be in public entry point.